sonic_installer/bootloader/grub.py

```diff
@@ -81,6 +85,36 @@ def remove_image(self, image):
         run_command('grub-set-default --boot-directory=' + HOST_PATH + ' 0')
         click.echo('Image removed')
 
+    def verify_image_platform(self, image_path):
+        if not os.path.isfile(image_path):
+            return False
+
+        # Get running platform's ASIC
+        try:
+            version_info = device_info.get_sonic_version_info()
+            if version_info:
+                asic_type = version_info['asic_type']
```

Comment:
>version_info['asic_type'] [](http://example.com/codeflow?start=28&length=25)

This not "running platform's ASIC". It is actually the current image's target ASIC.

For example:
1. considering you already force install a broadcom image onto a mellanox platform.
2. and in the running sonic (it is not working well), you want to install a next mellanox image without force
3. version info will tell you current asic is broadcom, and actually the running system ASIC is mellanox.